js/components/operatorSignIn/expiry.tsx

```diff
@@ -0,0 +1,163 @@
+import { daysBetween } from "../../dateTime";
+import {
+  Certification,
+  humanReadableType,
+  isExpired,
+} from "../../models/certification";
+import { className } from "../../util/dom";
+import { WarningParagraph } from "./warningParagraph";
+import { DateTime } from "luxon";
+import { ReactElement } from "react";
+
+const WARN_WITHIN_D = 60;
+
+const englishDaysBetween = (now: DateTime, date: DateTime) => {
+  const delta = daysBetween(now, date);
+  const wholeAbs = Math.abs(Math.ceil(delta));
+  if (delta <= -2) {
+    return `expired ${wholeAbs} days ago`;
+  } else if (delta <= -1) {
+    return "expired yesterday";
+  } else if (delta <= 0) {
+    return "expired today";
+  } else if (delta <= 1) {
+    return "expires tomorrow";
+  } else {
+    return `expires in ${wholeAbs} days`;
+  }
+};
+
+const CertificateBox = ({
+  now,
+  mode,
+  title,
+  operatorName,
+  certifications,
+}: {
+  now: DateTime;
+  mode: "warning" | "error";
+  title: string;
+  operatorName: string;
+  certifications: Certification[];
+}): ReactElement => {
+  const innerString =
+    `Our records show that ${operatorName}'s ` +
+    certifications
+      .map(
+        (c) =>
+          `${humanReadableType(c.type)} ${englishDaysBetween(
+            now,
+            c.expires,
+          )} on ${c.expires.toLocaleString(DateTime.DATE_SHORT)}`,
+      )
+      .join(" and ") +
+    ".";
+  return (
+    <WarningParagraph
+      className={className([
+        "border-0 light:bg-opacity-40 dark:bg-opacity-30 dark:text-white",
```

Comment:
refactoring suggestion: This is the only use of `WarningParagraph`, and these styles don't need to be controlled dynamically, so should they be part of `WarningParagraph` instead of passed in as an argument?

And should the background color be controlled by passing in `"warning"` or `"error"` instead of by giving an extraClassName?

Or, alternatively, WarningParagraph is just 2 divs and some styles, and it's not used anywhere else, so maybe it should be inlined here instead of abstracted?

Reply:
Inlined 👍 yeah, I honestly don't know why I made this separate in glides (which I copypastaed over here!)